plugins/plugin-java/che-plugin-java-ext-debugger-java-client/src/main/java/org/eclipse/che/ide/ext/java/jdi/client/debug/DebuggerServiceClient.java

```diff
@@ -27,129 +29,80 @@
  *
  * @author Vitaly Parfonov
  */
-public interface DebuggerServiceClient {
+public interface DebuggerServiceClient<T extends DebuggerConnectionContext>  {
+
     /**
      * Attach debugger.
-     *
-     * @param host
-     * @param port
-     * @param callback
      */
-    void connect(@NotNull String host, int port, @NotNull AsyncRequestCallback<DebuggerInfo> callback);
+    Promise<JavaDebuggerInfo> connect(@NotNull T connectionContext);
 
     /**
      * Disconnect debugger.
-     *
-     * @param id
-     * @param callback
      */
-    void disconnect(@NotNull String id, @NotNull AsyncRequestCallback<Void> callback);
+    Promise<Void> disconnect(@NotNull String id);
 
     /**
      * Adds breakpoint.
-     *
-     * @param id
-     * @param breakPoint
-     * @param callback
      */
-    void addBreakpoint(@NotNull String id, @NotNull BreakPoint breakPoint, @NotNull AsyncRequestCallback<Void> callback);
+    Promise<Void> addBreakpoint(@NotNull String id, @NotNull BreakPoint breakPoint);
 
     /**
-     * Returns list of breakpoints.
-     *
-     * @param id
-     * @param callback
+     * Deletes breakpoint.
      */
-    void getAllBreakpoints(@NotNull String id, @NotNull AsyncRequestCallback<String> callback);
+    Promise<Void> deleteBreakpoint(@NotNull String id, @NotNull BreakPoint breakPoint);
 
     /**
-     * Deletes breakpoint.
-     *
-     * @param id
-     * @param breakPoint
-     * @param callback
+     * Returns list of active breakpoints.
      */
-    void deleteBreakpoint(@NotNull String id, @NotNull BreakPoint breakPoint, @NotNull AsyncRequestCallback<Void> callback);
+    Promise<BreakPointList> getAllBreakpoints(@NotNull String id, @NotNull AsyncRequestCallback<String> callback);
 
     /**
      * Remove all breakpoints.
-     *
-     * @param id
-     * @param callback
      */
-    void deleteAllBreakpoints(@NotNull String id, @NotNull AsyncRequestCallback<String> callback);
+    Promise<Void> deleteAllBreakpoints(@NotNull String id);
 
     /**
      * Checks event.
-     *
-     * @param id
-     * @param callback
      */
     void checkEvents(@NotNull String id, @NotNull AsyncRequestCallback<DebuggerEventList> callback);
```

Comment:
It would be nice to create also promiseable version of this method, not now, but in nearest future.